libscript/src/script-builder.cpp

```diff
@@ -204,10 +204,40 @@ void MCScriptAddExportToModule(MCScriptModuleBuilderRef self, uindex_t p_definit
     }
 }
 
+void MCScriptAddImportToModuleWithIndex(MCScriptModuleBuilderRef self, uindex_t p_module_index, MCNameRef p_name, MCScriptDefinitionKind p_kind, uindex_t p_type_index, uindex_t p_def_index)
+{
+    uindex_t t_imp_index;
+    if (!__extend_array(self, self -> module . imported_definitions, self -> module . imported_definition_count, t_imp_index) ||
+        !MCMemoryNew((MCScriptExternalDefinition*&)self -> module . definitions[p_def_index]))
+    {
+        self -> valid = false;
+        return;
+    }
+    
+    __assign_definition_name(self, p_def_index, p_name);
+    
+    MCScriptImportedDefinition *t_import;
+    t_import = &self -> module . imported_definitions[t_imp_index];
+    
+    t_import -> module = p_module_index;
+    t_import -> kind = p_kind;
+    t_import -> name = MCValueRetain(p_name);
+    // t_import -> type = MCValueRetain(p_type);
```

Comment:
I've removed this field and the commented code - it dates back to before types were abstracted to structures in the module and so needs doing in a different way now anyway,